modules/test/playwright/pages/object-web/ModelBuilderPage.ts

```diff
@@ -92,6 +93,9 @@ export class ModelBuilderPage {
 			.getByRole('region')
 			.filter({has: page.getByTitle('Go to Folder')});
 		this.page = page;
+		this.rightSidebar = page.locator(
+			'div.lfr-objects__model-builder-right-sidebar-object-definition-node-details'
```

Comment:
Hi @rafaellajordao, in general we try not to use className in locators. Since we can easily rename the classes in the future. To help us deal with these locators, we can refer to [Playwright Quick Quide for Locators](https://playwright.dev/docs/locators#quick-guide).

In last cases we can opt for the uses of classes, but they should preferably be avoided.

In general, I would follow the following thought.

The effort required to modify a class is very small, this is something that can happen with some frequency.

However, there are some characteristics that tend to persist over time. For example, the rightsidebar would hardly stop being a sidebar.

Therefore, we can use this information to create our locators, for example.

We can say that the rightsidebar is a sidepanel element. However, we will have two elements of this type on this page, and we will need to filter the desired element.

A striking feature in the leftsidebar is the presence of the button, "create new object", and we can use this information to create a more precise selector. In other words, create a selector that captures the html elements with the role tabpanel. This would probably return two elements, then we would filter for the element that doesn't have the "create new object" button.

Finally, we can have something similar to this:

`page.getByRole("tabpanel").filter({hasNot: this.createNewObjectDefinitionButton})
`

Reply:
Thank you very much for this explanation @igor-franca , it is clearer to me now! :smile: